tests/tasks/test_control_flow.py

```diff
@@ -164,6 +164,107 @@ def test_merge_imperative_flow():
     assert state.result[c].result == 2
 
 
+def test_mapped_switch_and_merge_functional():
+    with Flow("iterated map") as flow:
+        mapped_result = identity.copy().map(["a", "b", "c"])
+
+        a = identity.copy().bind("a")
+        b = identity.copy().bind("b")
+        c = identity.copy().bind("c")
+
+        switch(condition=mapped_result, cases=dict(a=a, b=b, c=c), mapped=True)
+
+        merge_result = merge(a, b, c, flow=flow, mapped=True)
+
+    state = flow.run()
+
+    assert state.result[a].result == ["a", None, None]
+    assert state.result[b].result == [None, "b", None]
+    assert state.result[c].result == [None, None, "c"]
+    assert state.result[merge_result].result == ["a", "b", "c"]
+
+
+def test_mapped_switch_and_merge_imperative():
```

Comment:
I don't actually think this test (and the `test_mapped_ifelse_and_merge_imperative` test) add any extra coverage - just testing with the standard functional api here seems fine to me. No harm in having them, but if you feel like making the change it would shorten our test suite a bit.